welder.py

```diff
@@ -0,0 +1,50 @@
+import subprocess, os
+from flask import (Flask, request, abort, jsonify)
+
+app = Flask(__name__)  # create the application instance
+
+
+@app.route('/welder/api/syncTo', methods=['POST'])
+def sync_to():
+    if not request or 'destination' not in request.json:
+        abort(400)
+    destination = request.json['destination']
+    source = os.getenv('WELDERVOLUME','')
+    if not validate_bucket(destination):
+        return jsonify({"Destination is not a valid bucket":destination}), 400
+    if (source == ""):
+        return "Source not defined", 400
+    command = ['gsutil', '-m', 'rsync', '-r', source, destination]
+    output = ""
+    for line in run_command(command):
+        output += line 
```

Comment:
try `output = "separator".join(array)` instead of a loop here

Reply:
though what you're doing is building a giant string in memory... maybe you want to write the output to a file, and then find a way to stream that file into the response.